### PR TITLE
feature/ICKY-356-Checkout

### PR DESCRIPTION
**Pull request by kibo-sushant**

### CODE REVIEW

File: pages/cart.vue [hunk 1]
```diff
@@ -88,6 +89,8 @@
 import { SfButton, SfImage, SfHeading, SfBreadcrumbs, SfInput } from "@storefront-ui/vue"
 import { useAsync } from "@nuxtjs/composition-api"
 import { defineComponent } from "@vue/composition-api"
+import { useNuxtApp } from "#app"
```
Comment (seven-dot-k) [inline pages/cart.vue]: useRouter to push the path change instead

File: lib/gql/mutations/checkout/updateOrder.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import {
+  baseCheckoutFragment,
+  checkoutLineItemFragment,
+  fullfillmentInfoFragment,
+} from "../../fragments"
+
+export const updateOrder = /* GraphQL */ `
+  mutation updatePersonalDetails($orderId: String!, $updateMode: String, $orderInput: OrderInput) {
+    checkout: updateOrder(orderId: $orderId, updateMode: $updateMode, orderInput: $orderInput) {
+      email
```
Comment (seven-dot-k) [inline lib/gql/mutations/checkout/updateOrder.ts]: add email to baseCheckoutFragement

File: components/KiboOrderSummary.vue [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            const qty = current.qty
+            const price = current.price.special ? current.price.special : current.price.regular
+            const total = qty * parseFloat(price?.replace("$", ""))
+            return previous + total
+          }, 0)
+        : 0
+      return "$" + subtotal.toFixed(2)
+    })
+
+    const total = computed(() => {
```
Comment (seven-dot-k) [inline components/KiboOrderSummary.vue]: We need to create getters for all of these values, we never want to calculate ourselves

File: composables/useCheckout.ts [hunk 4]
```diff
@@ -38,10 +42,26 @@ export const useCheckout = () => {
...
 
-  const setShippingInfo = async (variables) => {
+  const setPersonalInfo = async (variables) => {
     loading.value = true
     try {
       const response = await fetcher({
+        query: updateOrder,
+        variables,
+      })
+      checkout.value = response?.data?.checkout
```
Comment (seven-dot-k) [inline composables/useCheckout.ts]: call `await load(checkout.value.id)` after the mutation instead

File: pages/cart.vue (hunk 1 above)
Comment (kibo-sushant) [reply pages/cart.vue]: using useRouter() now.

File: lib/gql/mutations/checkout/updateOrder.ts (hunk 2 above)
Comment (kibo-sushant) [reply lib/gql/mutations/checkout/updateOrder.ts]: added email in baseCheckoutFragement

File: components/KiboOrderSummary.vue (hunk 3 above)
Comment (kibo-sushant) [reply components/KiboOrderSummary.vue]: created checkoutGetters and using it

File: composables/useCheckout.ts (hunk 4 above)
Comment (kibo-sushant) [reply composables/useCheckout.ts]: now calling await load(checkout.value.id) after settingup personalInfo

**Review by kibo-kevinwatts — APPROVED** [review]

**Comment by kibo-kevinwatts** [comment]
This looks good!

**Pull request merged**